test/config/pg_types.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const should = require('should');
+const pgTypes = require('pg').types;
+const customTypes = require('../../config/pg_types');
+
+describe('parseDate', () => {
+  const DATE_OID = 1082;
+
+  it('does not offset time', () => {
+    let testDate = '2016-12-01';
+    let expectedDate = new Date(testDate).toISOString();
+    let resultedDate = pgTypes.getTypeParser(DATE_OID)(testDate);
+    resultedDate.toISOString().should.equal(expectedDate);
+  });
+
+  it('returns null when value is null', () => {
+    let testDate = null;
+    should.equal(pgTypes.getTypeParser(DATE_OID)(testDate), null);
+  });
+
+  it('raises when value is invalid', () => {
+    let testDate = 'not a date';
+    pgTypes.getTypeParser(DATE_OID).bind(null, testDate).should.throw(/^Invalid Date/);
```

Comment:
I used `.bind()` here because if I just do `pgTypes.getTypeParser(DATE_OID)(testDate).should.throw(/^Invalid Date/)` the tests fail with this error:

```
  1) parseDate raises when value is invalid:
     Error: Invalid Date
      at parseDate (config/pg_types.js:9:650)
      at Context.it (test/config/pg_types.js:25:36)
```